beets/autotag/mb.py

```diff
@@ -285,6 +289,27 @@ def album_info(release):
     artist_name, artist_sort_name, artist_credit_name = \
         _flatten_artist_credit(release['artist-credit'])
 
+    ntracks = sum(len(m['track-list']) for m in release['medium-list'])
+
+    # The MusicBrainz API omits 'artist-relation-list' and 'work-relation-list'
+    # when the release has more than 500 tracks. So we use browse_recordings
+    # on chunks of tracks to recover the same information in this case.
+    if ntracks > BROWSE_MAXTRACKS:
+        log.info(u'Album ' + str(release['id']) + u' has too many tracks')
+        log.info(u'Fetching recordings in batches of ' + str(BROWSE_CHUNKSIZE))
+        recording_list = []
+        for i in range(0, ntracks, BROWSE_CHUNKSIZE):
+            recording_list.extend(musicbrainzngs.browse_recordings(
+                    release=release['id'], limit=BROWSE_CHUNKSIZE,
+                    includes=BROWSE_INCLUDES,
+                    offset=BROWSE_CHUNKSIZE * i)['recording-list'])
+        for medium in release['medium-list']:
+            for recording in medium['track-list']:
+                recording_info = list(filter(lambda track: track['id'] ==
+                                             recording['recording']['id'],
+                                             recording_list))[0]
```

Comment:
A list comprehension might be a bit easier to read… `[track for track in recording_list if track['id'] == ...]`.

Or even better would be to first create a dict, `track_map`, mapping from track IDs to tracks, and then just have something like `track_map[recording['recording']['id']]`. That would make sure this remains O(n) rather than re-scanning the list each time, which could actually matter when we have 500+ tracks.

Reply:
Will do. This is really a failsafe measure, normally we could just take the recordings in the order they are given.